tests/beta/test_beta_qnode.py

```diff
@@ -1031,9 +1031,8 @@ def circuit(x):
         res = qml.grad(qml.grad(circuit))(x)
         assert np.allclose(res, -9 * np.cos(3 * x))
 
-    def test_hamiltonian_expansion_analytic(self, mocker):
-        """Test that the Hamiltonian is not expanded if there
-        are non-commuting groups and the number of shots is None"""

```

Comment:
@josh146 the following four files link to something I learned recently: don't use mocks that test which path a computation went (as long as it produces the desired result). I like this logic: the tests are our promise, and we do not promise (yet) how Hamiltonians are evaluated, just that they are evaluated correctly. So I hope it's fine if I simply delete!

Reply:
> I like this logic: the tests are our promise, and we do not promise (yet) how Hamiltonians are evaluated, just that they are evaluated correctly.

Makes a lot of sense, and will lead to easier to maintain/more robust tests :)